NodeApi/JSPropertyDescriptorListOfT.cs

```diff
@@ -0,0 +1,188 @@
+using System;
+using System.Collections.Generic;
+
+namespace NodeApi;
+
+public class JSPropertyDescriptorList<TDerived, TObject>
+  where TDerived : class, IJSObjectUnwrap<TObject>
+  where TObject : class
+{
+  public IList<JSPropertyDescriptor> Properties { get; } = new List<JSPropertyDescriptor>();
+
+  public JSPropertyDescriptorList() { }
+
+  public TDerived AddProperty(
+    string name,
+    JSValue value,
+    JSPropertyAttributes attributes = JSPropertyAttributes.DefaultProperty)
+  {
+    Properties.Add(new JSPropertyDescriptor(name, value, attributes));
+    return (TDerived)(object)this;
+  }
+
+  public TDerived AddProperty(
+    string name,
+    JSCallback? getter,
+    JSCallback? setter,
+    JSPropertyAttributes attributes = JSPropertyAttributes.DefaultProperty)
+  {
+    Properties.Add(new JSPropertyDescriptor(name, getter, setter, attributes));
```

Comment:
For all callbacks from JS to C#, including property getters and setters and methods, I think there needs to be some code that catches any exception thrown by the C# callback and converts it to a JS error. (I had a TODO for that in the [`Object.InvokeCallback()`](https://github.com/jasongin/napi-dotnet/blob/main/NodeApi/Object.cs#L193) code deleted in this PR.

Reply:
I am adding a TODO for now.